lib/src/components/NxFilterInput/NxFilterInput.tsx

```diff
@@ -23,16 +26,43 @@ const NxFilterInput = forwardRef<HTMLDivElement, Props>(
           className = classnames('nx-filter-input', classNameProp, {
             'nx-filter-input--empty': isEmpty
           }),
-
+          btnClassName = classnames('nx-btn--clear', { 'hidden': isEmpty }),
+          fieldRef = useRef<HTMLDivElement>(null),
+          mergedRef = useMergedRef(fieldRef, ref),
           // just in case these props get passed in, avoid passing them to NxTextInput as they would cause
           // malfunction
           cleanedProps = omit(['validatable', 'validationErrors', 'type'], otherProps),
           filterIcon = searchIcon ? faSearch : faFilter,
-          prefixContent = <NxFontAwesomeIcon icon={filterIcon} className="nx-icon--filter-icons" />;
+          prefixContent = <NxFontAwesomeIcon icon={filterIcon} className="nx-icon--filter-icons" />,
+          suffixContent =
+            <NxButton className={btnClassName}
+                      variant="icon-only"
+                      title="clear filter"
+                      onClick={clearFilterInputText}
+                      tabIndex={-1}><Close/>
+            </NxButton>;
+
+      function clearFilterInputText() {
```

Comment:
This seems way over-complicated.  Is there a reason you can't just call the `onChange` prop with `''`?

Reply:
> But then I found a way to avoid the type warning https://github.com/sonatype/sonatype-react-shared-components/commit/4c7a67337521345304121c8fc8bb88c051e2bd83, not sure if this is okay, please let me know your opinions, thanks.

Using `partial` there to just break the typechecking isn't good, a more straightforward way to do the same thing would be to just cast to `any`.  But of course we'd rather not break the typechecking.  It's interesting that the type for `onChange` here doesn't match what's described in the documentation.  The type has it taking two parameters, the string and the event, while the gallery docs make no mention of the event.  Since the docs are silent about it anyway, I think we can change the type to indicate that the second parameter is optional.

Reply:
Thanks for the opinions and suggestions! Here is the updates 06843871f8a0937f619868f15845bf540ccd0893.